### PR TITLE
Condition tutorial on active HMD

### PR DESCRIPTION
**Pull request by zzmp**
- Changes Interface start logic to only teleport to tutorial if HMD is in use
- Changes tutorial entity script to only start tutorial if HMD.active

### CODE REVIEW

**Comment by zzmp** [comment]
This needs some QA setup, so hold off on QA until I've put the tag up.

**Comment by hifi-gustavo** [comment]
Build console output for commit: f432cfea6292098fc4c495fdc45c5f7935d26b98 available at: https://gist.github.com/9723bc0919dc40497463d8d89f8c7eb7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10240-f432cfe.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10240-f432cfe.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -1458,46 +1458,53 @@ Application::Application(int& argc, char** argv, QElapsedTimer& startupTimer, bo
...
+            handControllerType = Oculus;
         }
 
-        Setting::Handle<bool> firstRun { Settings::firstRun, true };
+        // Check tutorial content versioning
+        bool hasTutorialContent = contentVersion >= MIN_CONTENT_VERSION.at(handControllerType);
+
+        // Check HMD use (may be technically available without being in use)
+        bool hasHMD = PluginUtils::isHMDAvailable();
+        bool isUsingHMD = hasHMD && hasHandControllers && _displayPlugin->isHmd();
```
Comment (huffman) [inline interface/src/Application.cpp]: To confirm, we are now booting directly into the HMD on a fresh install if an HMD is available? And we will have switched to it by the time we get here?

Comment (zzmp) [inline interface/src/Application.cpp]: I haven't "confirmed" this in codepaths, but I have noted this in all of my testing.

Comment (zzmp) [inline interface/src/Application.cpp]: I will look through codepaths right now to confirm. I'm fairly certain this happens for Vive (booting from Steam), not as sure about Oculus.

Comment (zzmp) [inline interface/src/Application.cpp]: Yes. We do not boot directly into an HMD, we boot into the first menu item. However, Application::loadSettings will switch to an HMD on first run, before this check (for tutorial) runs.

This occurs [here](https://github.com/zzmp/hifi/blob/f432cfea6292098fc4c495fdc45c5f7935d26b98/interface/src/Application.cpp#L3779).

**Comment by zzmp** [comment]
### QA Plan

In all tests, you should be running a local server.

#### Test that you are put into the correct location when you start Interface.

On a fresh install, you should always start in HMD mode if possible.
To default to desktop, unplug your HMD so that it fails to engage.

| Start condition | Start location |
| :- | :- |
| HMD Mode, with Hand Controllers, Not finished tutorial | Tutorial Area |
| HMD Mode, Finished tutorial | Last location |
| HMD Mode, sandbox not running | Last location or Welcome, if first run |
| Desktop Mode, Not finished tutorial | Welcome |
| Desktop Mode, Finished tutorial | Last location |
| Desktop Mode, sandbox not running | Last location or Welcome, if first run |

#### Test that the tutorial only starts if you are in HMD.

- Go to your local server in Desktop Mode.
- Edit the `tutorial/start_container` by changing its `Script URL` to https://raw.githubusercontent.com/zzmp/hifi/f432cfea6292098fc4c495fdc45c5f7935d26b98/tutorial/tutorialStartZone.js (the script in this PR).
- Go to `Restart Tutorial` teleporter.
- You should be unable to do the tutorial. A dialog should open up telling you that you need to be in HMD.
- Go back and `Restart Tutorial` in HMD. It should work.

**Comment by OmegaHeron** [comment]
Testing this one.

**Comment by OmegaHeron** [comment]
HMD Mode, with Hand Controllers, Not finished tutorial - Yes, started at tutorial and was able to complete.
HMD Mode, Finished tutorial - Yes, started at last location.
HMD Mode, sandbox not running - Yes, started at last location for case of having finished tutorial and welcome for case of sandbox not running.

Desktop Mode, Not finished tutorial - Attempted to start in dev-welcome, proto mismatch given (pass none the less as it took me to correct place even if it couldn't finish deal).
Desktop Mode, Finished tutorial - Yes, returned to last location.
Desktop Mode, sandbox not running - Yes, returned to last location or (dev-)welcome on first run.

There's still an issue, likely not really Interface's, where you can take the following process;
Unplug Oculus CV1
Unplug all USB position readers
Remove batteries from both touch controllers
Restart PC
Start Interface from a clean install (appdata stuff removed)
Note Interface starts in desktop mode and goes to dev-welcome presenting "tutorial" graphic
Steam VR creates to dialogs; 1) Headset not ready 2) SteamVR not ready
Oculus Runtime application launches giving warning messages that hardware is not attached
Quit Interface
Close Oculus application
Dismiss Steam dialogs/close steam vr from tray icon
Launch Interface again
You return to last location, you do not get tutorial graphic, you do, once again, launch Oculus application then Steam VR.

Understand all below;
I'm not running this from steam
I have *all* my gear unplugged and have rebooted computer
It seems interface, maybe, is seeing oculus service running or "something" that then triggers oculus runtime application to load then it fails and here comes steam VR.
I'm willing, totally, to accept this is just the way it is, but, *only* if this a "feature" vs bug -- otherwise it's totally annoying.

**Comment by zzmp** [comment]
> Note Interface starts in desktop mode and goes to dev-welcome presenting "tutorial" graphic

The "tutorial" graphic should be happening on localhost, not dev-welcome. What is happening?

For the "misbehaving" case, can you include your logs? I'd like to see whether the HMD and hand controllers were detected.

**Comment by OmegaHeron** [comment]
If sandbox !running and first run then goto dev-welcome and present graphic.  I thought that was correct.  If sandbox running it goes to sandbox, presenting graphic on first run.

I'll upload logs to slack tomorrow - will need to run test again after removing all HMD gear from system.  I've already uninstalled PR and removed its appData folders.

**Comment by OmegaHeron** [comment]
Logs uploaded to slack @zzmp

**Pull request merged**